### PR TITLE
Add categorical from pandas df

### PR DESCRIPTION
**Pull request by samihamdan**
Adding that categorical features are detected by run_cross_validate and treated appropriately. (#125 )

### CODE REVIEW

**Comment by samihamdan** [comment]
Tests have to be added and adjusted but basic functionality was already added.

File: julearn/api.py [hunk 1]
```diff
@@ -217,6 +215,7 @@ def run_cross_validation(
 def create_pipeline(
     model,
     confounds=None,
+    categorical_features=None,
```
Comment (fraimondo) [inline julearn/api.py:218]: Why do we need this here? The pipeline should not care about the input data. This is something that is dealt with in the `fit` method.

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: julearn/api.py (hunk 1 above)
Comment (kaurao) [reply julearn/api.py:218]: perhaps to auto hot-encode them?

**Comment by fraimondo** [comment]
We are avoiding all sort of automatic behaviour.


I think the problem is bigger than this PR and its also related to confounds.

SKLearn pipelines do not have, when not fitted, any information about the features. Julearn pipelines can have information about the confounds. That makes a julearn pipeline not suitable to be applied directly in other domain.

Both `confounds` and `categorical_features` should be a parameter of the `fit` methods and not of the constructor.

File: julearn/api.py (hunk 1 above)
Comment (samihamdan) [reply julearn/api.py:218]: We did not yet add auto hot-encoding. 
But this should also be able as a feature for `run_cross_validation` if we put `categorical_features` into `fit_params`. 
I am not sure why we did not do that before, but I think we were not completely sure how to do it best and then I followed what we already had. 
I will try to implement both `categorical_featrues` and `confounds` into `fit_params` if possible in #142 .

**Comment by samihamdan** [comment]
The functionality of this PR should be covered by #142.

**Pull request closed without merging**